nav-tracking-mitigations/stateful-client-bounce.sub.https.html

```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE HTML>
+<meta charset=utf-8>
+<title>Bounce Tracking Mitigations: Stateful Client Bounce</title>
+<body>body for test_driver.bless</body>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script src=/resources/testdriver.js></script>
+<script src=/resources/testdriver-vendor.js></script>
+<script>
+  promise_setup(async () => {
+    try {
+      await test_driver.set_storage_access('*', '*', 'blocked');
+    } catch (e) {
+      // Ignore, can be unimplemented if the platform blocks cross-site
+      // cookies by default. If this failed without default blocking the test
+      // will fail.
+    }
+  });
+
+  promise_test(async t => {
+    const {promise: getLinkPromise, resolve: getLinkResolver} =
+      Promise.withResolvers();
+
+    // Open a new tab and retrieve the link to start the test.
+    await test_driver.bless('open page to start test', async () => {
+      const otherTab = window.open(
+        'resources/redirect-chain-start.sub.https.html');
+
+      await new Promise(resolve => otherTab.addEventListener('load', resolve));
+      getLinkResolver(otherTab.document.getElementById('navigate-link'));
+    });
+
+    // Click the link in the new tab.
+    //
+    // Ideally, we would click this from within the page, but because the page
+    // immediately navigates away, test_driver fails as it expects the page to
+    // not navigate away. Doing this from the main test page avoids the issue.
+    const link = await getLinkPromise;
+    test_driver.click(link);
+
+    await new Promise(resolve => window.addEventListener('message', resolve));
+
+    let result = [];
+    while (result.length == 0) {
+      result = await test_driver.run_bounce_tracking_mitigations();
+    }
```

Comment:
Why is a `while` loop required here? Is it because the bounce tracker classification only runs after a delay (on extended navigation end)? Right now that delay is implementation defined, but the spec recommends 10 seconds. 
Even if we're okay with waiting for 10 seconds, we should not spin the CPU for that long here. We can either add wait or make`run_bounce_tracking_mitigations` skip the 10 second wait.

The other way for an extended navigation to end (as per the spec) is if the user navigates again (before the 10 second timeout). We probably want to test both cases. I don't want to require it here since this is just the first basic bounce tracking mitigation test.

Reply:
How does the [extended navigation](https://privacycg.github.io/nav-tracking-mitigations/#extended-navigation) in the test end? From my understanding of the spec (and please correct me if I'm wrong) either it ends by [timeout](https://privacycg.github.io/nav-tracking-mitigations/#client-bounce-detection-timer-period) or because a new navigation is initiated by the user. After the bounce I don't see another navigation triggered by the "user". Wouldn't we have to wait at least 10 seconds until the bounce tracker is even added to the stateful bounce tracking map?

Reply:
Oh, I see. We don't have that behavior in Firefox currently. We discard the bounce tracking record when the top level traversable gets destroyed.  I think it might be a gap in the spec.
I'm fine with landing this test as-is and moving the spec discussion over to https://github.com/privacycg/nav-tracking-mitigations